### DISCUSSION TITLE
How does Orchard Core register IShellConfiguration to be resolved as ShellConfiguration concrete type?

### DISCUSSION THREAD

**Discussion by aaronamm** [OrchardCMS/OrchardCore#13745] [Q&A] [answered]
Hello everyone. 

I have a simple question but I can't figure it out by myself. 
**How does Orchard Core register IShellConfiguration to be resolved as ShellConfiguration concrete type?**
I could not find anywhere in source code to show how IShellConfiguration is registered as singleton.

This question is started from what I couldn't get IShellConfiguration to read my custom configuration in appsettings.json.

Actually, a custom configuration needs to be inside OrchardCore section/object because configuration is read from OrchardCore section which is defined in 
https://github.com/OrchardCMS/OrchardCore/blob/main/src/OrchardCore/OrchardCore/Shell/ShellSettingsManager.cs#L277

I hope if you can help me find where IShellConfiguration get registered. 
Thanks.

**Comment by jtkech** [comment, marked as answer]
https://github.com/OrchardCMS/OrchardCore/blob/699fcaf9607367950f4cd07676651283305971a5/src/OrchardCore/OrchardCore/Shell/Builders/ShellContainerFactory.cs#L39-L45

**Reply by aaronamm** [reply]
@jtkech, thank you so much.
